python/lsst/pipe/base/graphBuilder.py

```diff
@@ -658,17 +658,27 @@ def fillQuanta(self, registry, inputCollections, *, skipExisting=True):
                 # a different calibration_label, or a refcat with a skypix
                 # value that overlaps the quantum's data ID's region, but not
                 # the user expression used for the initial query.
-                for datasetType, scaffolding in task.prerequisites.items():
-                    refs = list(
-                        registry.queryDatasets(
-                            datasetType,
-                            collections=inputCollections[datasetType.name],
-                            dataId=quantumDataId,
-                            deduplicate=True,
-                            expand=True,
+                connections = task.taskDef.connections
+                for con_name in connections.prerequisiteInputs:
+                    con = getattr(connections, con_name)
+                    for datasetType, _ in task.prerequisites.items():
```

Comment:
and usually not the `.keys()` if `task.prerequisites` is `dict`-like.

Reply:
`.keys()` is clearer, IMO.

Reply:
Hmm. We went through all the code and removed the `.keys()` usage during the migration to python3.  My impression is that this decision was specifically to allow cleaner looping over keys.   `dict` iterators are the keys of the `dict`.

Reply:
:slightly_frowning_face:  Which decision?  Python's change to what `keys` means, or our removal of `.keys` from our code?

Using `keys()` makes it clear that the iterable _is_ a dictionary (often important when types are not obvious), and I've always found it weird that `iter(dict)` doesn't yield items (obviously I'm used to it now, but I always figured that by Python 7.0 or so the language would finally figure that out as well).

Reply:
I have no opinion here, I just used the existing line of code and dropped the unused variable without thinking about it. My only statement would be the existence of certain methods does not imply the type of the object, as we have way too many mapping like objects, and at a glace there is no easy way to know which of the mapping types it may be, and which methods of the mapping interface they may or may not have. This means the type must be looked up anyway (or just test to see).